eth/plasma.go

```diff
@@ -201,5 +198,42 @@ func (plasma *Plasma) HasTxBeenExited(position plasmaTypes.Position) bool {
 		return true
 	}
 
-	return e.State == 1 || e.State == 3
+	ethBlockNum, err := plasma.ethBlockPeg(tmBlock)
+	if err != nil {
+		plasma.logger.Error(fmt.Sprintf("Could not get associated ETH Block for sidechain block %d: %s", tmBlock.Int64(), err.Error()))
+		return true
+	}
+
+	// Return true if exit is pending/finalized AND exit happened before the pegged ETH block
+	return (e.State == 1 || e.State == 3) && e.EthBlockNum.Cmp(ethBlockNum) == -1
```

Comment:
This should be `<=0` right? B/c if the exit is pending or finalized at the same eth block being used then the spend of it should fail